### PR TITLE
Make the docs more user friendly

### PR DESCRIPTION
**Pull request by 4383**
Improve the general index by adding a contribution guide, changelog, etc.

Refactor sections to host various topics and to generate a more sexy table of context which will be more user friendly.

Refactor formatting heading levels.

Make Asyncio warning message reusable and centralized.

Give more visibility to the Asyncio topic.

### CODE REVIEW

File: doc/source/asyncio/compatibility.rst [hunk 1]
```diff
@@ -1,4 +1,7 @@
-# Asyncio compatibility in eventlet
+.. _asyncio-compatibility:
```
Comment (4383) [inline doc/source/asyncio/compatibility.rst:1]: IMO it is better to move this file in the doc source and at the rst format.

File: doc/source/reference/api/eventlet.hubs.rst [hunk 2]
```diff
@@ -44,6 +44,14 @@ eventlet.hubs.poll module
    :undoc-members:
    :show-inheritance:
 
+eventlet.hubs.pyevent module
```
Comment (itamarst) [inline doc/source/reference/api/eventlet.hubs.rst:47]: Did you mean to restore this?

**Review by itamarst — APPROVED** [review]
Seems fine other than the pyevent thing.

File: doc/source/reference/api/eventlet.hubs.rst (hunk 2 above)
Comment (4383) [reply doc/source/reference/api/eventlet.hubs.rst:47]: Oh good catch. I think I forgot to rebase my development branch... Will fix that

**Pull request merged**